R/tm_outliers.R

```diff
@@ -246,52 +247,56 @@ srv_outliers <- function(id, datasets, reporter, outlier_var,
       }
     })
 
-    merged_data <- teal.transform::data_merge_srv(
+    anl_merged_input <- teal.transform::merge_expression_srv(
       selector_list = reactive_select_input,
-      datasets = datasets,
+      datasets = data,
+      join_keys = attr(data, "join_keys"),
       merge_function = "dplyr::inner_join"
     )
 
-    is_cat_filter_spec <- inherits(categorical_var[[1]]$filter[[1]], "filter_spec")
-    cat_dataname <- categorical_var[[1]]$dataname
-
-    common_code_chunks <- reactive({
-      validate(need(!is.null(reactive_select_input()$outlier()), "Please select an outlier variable dataset"))
```

Comment:
Red error when deselected:
![Screenshot from 2022-08-29 18-23-39](https://user-images.githubusercontent.com/86970066/187248388-bb25c28c-aed7-47d7-931a-b9fa9a290be3.png)

Reply:
similar error when deselecting categorical var:
![Screenshot from 2022-08-29 18-32-30](https://user-images.githubusercontent.com/86970066/187250183-e2936f48-9e82-4199-9876-33348e3a2052.png)

and method:
![Screenshot from 2022-08-29 18-33-06](https://user-images.githubusercontent.com/86970066/187250223-f54ee8fa-9c1b-40f0-899f-15c498a00c0a.png)

Reply:
fixed